src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,479 @@
+        }
+
+        // This method is registered by the CancellationTokenSource cts in the connect method
+        // and called by Dispose or Abort so that any open websocket connection can be closed.
+        private async void AbortRequest()
+        {
+            if (State == WebSocketState.Open)
+            {
+                await CloseAsyncCore(WebSocketCloseStatus.NormalClosure, SR.net_WebSockets_Connection_Aborted, CancellationToken.None).ConfigureAwait(continueOnCapturedContext: true);
+            }
+        }
+
+        /// <summary>
+        /// Send data on <see cref="System.Net.WebSockets.ClientWebSocket"/> as an asynchronous operation.
+        /// </summary>
+        /// <returns>The async.</returns>
+        /// <param name="buffer">Buffer.</param>
+        /// <param name="messageType">Message type.</param>
+        /// <param name="endOfMessage">If set to <c>true</c> end of message.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            if (messageType != WebSocketMessageType.Binary &&
+                    messageType != WebSocketMessageType.Text)
+            {
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_Argument_InvalidMessageType,
+                    messageType,
+                    nameof(SendAsync),
+                    WebSocketMessageType.Binary,
+                    WebSocketMessageType.Text,
+                    nameof(CloseOutputAsync)),
+                    nameof(messageType));
+            }
+
+            WebSocketValidate.ValidateArraySegment(buffer, nameof(buffer));
+
+            _writeBuffer ??= new MemoryStream();
+            _writeBuffer.Write(buffer.Array!, buffer.Offset, buffer.Count);
+
+            if (!endOfMessage)
+                return Task.CompletedTask;
+
+            MemoryStream writtenBuffer = _writeBuffer;
+            _writeBuffer = null;
+
+            try
+            {
+                switch (messageType)
+                {
+                    case WebSocketMessageType.Binary:
+                        using (Uint8Array uint8Buffer = Uint8Array.From(buffer))
+                        {
+                            _innerWebSocket!.Invoke("send", uint8Buffer);
+                        }
+                        break;
+                    default:
+                        string strBuffer = buffer.Array == null ? string.Empty : Encoding.UTF8.GetString(buffer.Array, buffer.Offset, buffer.Count);
+                        _innerWebSocket!.Invoke("send", strBuffer);
+                        break;
+                }
+            }
+            catch (Exception excb)
+            {
+                return new Task.FromException(WebSocketException(WebSocketError.NativeError, excb));
```

Comment:
I think that's what @stephentoub meant:

```suggestion
                return Task.FromException(new WebSocketException(WebSocketError.NativeError, excb));
```

Reply:
Yup, typo.

Reply:
Yes, I already updated it.  Thanks